### DISCUSSION TITLE
Hard fault whwn calling imagePush

### DISCUSSION THREAD

**Discussion by mesbahuddin** [lovyan03/LovyanGFX#648] [Q&A]
Hello ...

When i run below code i get a hard fault at pushImage 

#define RGB_TEST_HEIGHT 6
#define RGB_TEST_WIDTH 4

// array size is 48
static const uint16_t rgb_test[]  = {
  0xf800, 0xf800, 0xf800, 0xf800,
  0xf800, 0xf800, 0xf800, 0xf800,
  0x07e0, 0x07e0, 0x07e0, 0x07e0,
  0x07e0, 0x07e0, 0x07e0, 0x07e0,
  0x001f, 0x001f, 0x001f, 0x001f,
  0x001f, 0x001f, 0x001f, 0x001f
};

int main(void) {
	
	auto canvas = lgfx::LGFX_Sprite();
	canvas.createSprite(RGB_TEST_WIDTH, RGB_TEST_HEIGHT);
	canvas.setColorDepth(lgfx::rgb565_nonswapped);
	
	canvas.pushImage(0, 0, RGB_TEST_WIDTH, RGB_TEST_HEIGHT, (uint16_t*)rgb_test);
	
	/* Infinite loop */
	while (1) {
		HAL_Delay(500);
	}
	
}

**Comment by tobozo** [comment]
hi,

the sprite instance isn't attached to anything, it should either be attached to the display when creating the object:

```cpp
   auto canvas = lgfx::LGFX_Sprite(&tft);
```

... or the target (can be display or another sprite) should be specified when pushing the sprite:

```cpp
    canvas.pushImage(&tft, 0, 0, 48, 48, (uint16_t*)rgb_test);
```

if you still get crashes after that then it's probably that RGB_TEST_WIDTH*RGB_TEST_HEIGHT is not equal to 24

**Comment by mesbahuddin** [comment]
Acyually i am trying to use teh library on an unsupported MCU ... STM32h7 with LTDC display ... my idea was to draw on a canvas than transfer teh canvas to the display using STM32 specific function calls .. i ven found a librarty taht already stripped hardware dependencies from lovyanGFX https://github.com/Forairaaaaa/lgfx_slim

**Reply by tobozo** [reply]
even in the slim version the constructor needs a `parent` argument which is either the panel object or another sprite

https://github.com/Forairaaaaa/lgfx_slim/blob/main/src/lgfx/v1/LGFX_Sprite.hpp#L108C1-L108C35
